### PR TITLE
Enhancement/better web component name validation

### PR DESCRIPTION
**Pull request by vilarfg**
By using Sindre Sorhus' [validate-element-name](https://www.npmjs.com/package/validate-element-name), we can make sure the element's name comply with the [specs](https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name).

 - [x] CHANGELOG.md has been updated

### CODE REVIEW

**Comment by vilarfg** [comment]
I signed it!

File: src/init/application/application.ts [hunk 1]
```diff
@@ -76,27 +75,27 @@ export function createApplicationGenerator(templateName: string):
 
       this.props = await this.prompt(prompts);
       this.props.elementClassName = this.props.elementName.replace(
-          /(^|-)(\w)/g,
-          (_match: string, _p0: string, p1: string) => p1.toUpperCase());
+        /(^|-)(\w)/g,
```
Comment (justinfagnani) [inline src/init/application/application.ts]: Please revert the whitespace only changes. Our style is 4 space indent on wrapping. Try running `npm run format`, if there are changes in code other than what you touched, put those in another commit.

File: src/init/application/application.ts [hunk 2]
```diff
@@ -58,13 +59,11 @@ export function createApplicationGenerator(templateName: string):
...
           default: (answers: any) => `${answers.name}-app`,
           validate: (name: string) => {
-            const nameContainsHyphen = name.includes('-');
-            if (!nameContainsHyphen) {
-              this.log(
-                  '\nUh oh, custom elements must include a hyphen in ' +
-                  'their name. Please try again.');
+            const validation = validateElementName(name);
+            if(!validation.isValid) {
+              this.log(`\nUh oh. ${validation.message} Please try again.`);
```
Comment (justinfagnani) [inline src/init/application/application.ts]: I think you can remove the "Uh oh" part and put "Please try again" on a new line.

File: src/init/element/element.ts [hunk 3]
```diff
@@ -53,13 +54,11 @@ export function createElementGenerator(templateName: string):
...
               this.appname + (this.appname.includes('-') ? '' : '-element'),
           validate: (name: string) => {
-            const nameContainsHyphen = name.includes('-');
-            if (!nameContainsHyphen) {
-              this.log(
-                  '\nUh oh, custom elements must include a hyphen in ' +
-                  'their name. Please try again.');
+            const validation = validateElementName(name);
+            if(!validation.isValid) {
+              this.log(`\nUh oh. ${validation.message} Please try again.`);
```
Comment (justinfagnani) [inline src/init/element/element.ts]: Same comment here with the message.

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 ## Unreleased
 - Added support for optional polymer-project-config provision of bundler options instead of only boolean value for the `bundle` property of build definitions.
+- improved web component name validation by using Sindre Sorhus' validate-element-name.
```
Comment (justinfagnani) [inline CHANGELOG.md]: I don't think we need the developer's name in the change description.

**Comment by stramel** [comment]
Seems like this may be a good linter rule as well

**Comment by justinfagnani** [comment]
@stramel yeah. cc @rictic

**Comment by rictic** [comment]
Filed as https://github.com/Polymer/polymer-linter/issues/95

**Comment by FredKSchott** [comment]
@vilarfg bump! are you still able to work on this to get it over the line?

**Comment by vilarfg** [comment]
@justinfagnani @FredKSchott I think that's it! am I missing sth? cheers

**Comment by FredKSchott** [comment]
Closing this PR as abandoned. This happens if there hasn't been any activity on a PR in a long time.

@vilarfg if you are still interested in getting this PR merged: please rebase & respond to comments, and then ping me to reopen and I'll be happy to take a look.
